common/v2/services/Store/DataManager/actions.ts

```diff
@@ -0,0 +1,107 @@
+import { Dispatch } from 'react';
+import { SymmetricDifference } from 'utility-types';
+import {
+  LSKeys,
+  TUuid,
+  DataStore,
+  DataStoreEntry,
+  DataStoreItem,
+  Network,
+  NetworkId,
+  ISettings,
+  LocalStorage
+} from 'v2/types';
+import { ActionPayload, ActionV, ActionT } from './reducer';
+import { marshallState, deMarshallState } from './utils';
+
+type createPayload = (m: LSKeys) => <T>(v: T) => ActionPayload<T>;
+const createPayload: createPayload = model => v => ({ model, data: v });
+
+export function ActionFactory(model: LSKeys, dispatch: Dispatch<ActionV>, state: DataStore) {
+  const generatePayload = createPayload(model);
+
+  const create = (item: DataStoreItem): void => {
+    dispatch({
+      type: ActionT.ADD_ITEM,
+      payload: generatePayload(item)
+    });
+  };
+
+  const createWithID = (
+    item: SymmetricDifference<DataStoreItem, ISettings | Network>,
+    uuid: TUuid
+  ): void => {
+    dispatch({
+      type: ActionT.ADD_ITEM,
+      payload: generatePayload({ ...item, uuid })
+    });
+  };
+
+  // Ideally we would want to add a Union constraint to our Generic param.
+  // https://github.com/microsoft/TypeScript/issues/13995
+  // In the mean time we have to cast our id.
+  const update = <T>(
+    uuid: T,
+    item: T extends NetworkId
+      ? Network
+      : T extends TUuid
+      ? SymmetricDifference<DataStoreItem, Network>
+      : never
+  ): void => {
+    if (model === LSKeys.NETWORKS) {
+      dispatch({
+        type: ActionT.UPDATE_NETWORK,
+        payload: generatePayload({ ...item, id: (uuid as unknown) as NetworkId })
+      });
+    } else {
+      dispatch({
+        type: ActionT.UPDATE_ITEM,
+        payload: generatePayload({ ...item, uuid: (uuid as unknown) as TUuid })
+      });
+    }
+  };
+
+  const updateAll = (entry: DataStoreEntry): void => {
+    dispatch({
+      type: ActionT.ADD_ENTRY,
+      payload: generatePayload(entry)
+    });
+  };
+
+  const destroy = (item: DataStoreItem): void => {
+    dispatch({
+      type: ActionT.DELETE_ITEM,
+      payload: generatePayload(item)
+    });
+  };
+
+  const destroyStorage = () => {
+    console.debug('Destroy Storage');
+  };
+
+  const importStorage = (data: string) => {
+    const d = JSON.parse(data);
+    // @TODO: perfom version and validity check.
```

Comment:
Seems to be a problem with importing previous backups. It shows an error in the settings/import, but still imports i think?

May be due to this todo

Reply:
With the latest change to `isValidImport` we check that the imports `version` and `key` corresponds to the accepted db. This means that we fail immediately if the file is invalid.

For a better UX i would suggest we start using to https://github.com/epoberezkin/ajv to validate the schema. If we choose to do so we can start generating our types from the schema definition https://github.com/bcherny/json-schema-to-typescript, produce more specific errors and eventually be able to migrate between the different schemas. 
ie: https://spin.atomicobject.com/2018/03/26/typescript-data-validation/

Since this change would requires modifying all our data types I prefer doing in another chore.